src/admin/resolvers/queries/RejectedItem.auth.integration.ts

```diff
@@ -0,0 +1,145 @@
+    },
+    {
+      title: 'The Untold Secret To Mastering PHP In Just 3 Days',
+    },
+    {
+      title: 'You Can Thank Us Later - 3 Reasons To Stop Thinking About PHP',
+    },
+  ];
+
+  let headers: {
+    name: string | undefined;
+    username: string | undefined;
+    groups: string | undefined;
+  } = {
+    name: 'Test User',
+    username: 'test.user@test.com',
+    groups: undefined,
+  };
+
+  beforeAll(async () => {
+    await clearDb(db);
+
+    for (const item of rejectedCuratedCorpusItems) {
+      await createRejectedCuratedCorpusItemHelper(db, item);
+    }
+  });
+
+  afterAll(async () => {
+    await db.$disconnect();
+  });
+
+  describe('getRejectedCuratedCorpusItem query', () => {
+    it('should get all items when user has read-only access', async () => {
+      // Set up auth headers with read-only access
+      headers = {
+        ...headers,
+        groups: `this,that,${MozillaAccessGroup.READONLY}`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_REJECTED_ITEMS,
+      });
+
+      expect(result.errors).to.be.undefined;
+      expect(result.data).not.to.be.null;
+
+      // should return all four items - the entire corpus should be accessible
+      expect(result.data?.getRejectedCuratedCorpusItems.edges).to.have.length(
+        4
+      );
+
+      await server.stop();
+    });
+
+    it('should get all items when user has only one scheduled surface access', async () => {
+      // Set up auth headers with access to a single Scheduled Surface
+      headers = {
+        ...headers,
+        groups: `this,that,${MozillaAccessGroup.NEW_TAB_CURATOR_ENUS}`,
+      };
+
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_REJECTED_ITEMS,
+      });
+
+      expect(result.errors).to.be.undefined;
+      expect(result.data).not.to.be.null;
+
+      // should return all four items - the entire corpus should be accessible
+      expect(result.data?.getRejectedCuratedCorpusItems.edges).to.have.length(
+        4
+      );
+
+      await server.stop();
+    });
+
+    it('should throw an error when user does not have the required access', async () => {
+      // Set up auth headers with no valid access group
+      headers = {
+        ...headers,
+        groups: `this,that`,
+      };
+      const server = getServerWithMockedHeaders(headers);
+      await server.start();
+
+      const result = await server.executeOperation({
+        query: GET_REJECTED_ITEMS,
+      });
+
+      expect(result.data).to.be.null;
+      expect(result.errors).not.to.be.undefined;
+
+      // check if the error we get is the access denied error
+      expect(result.errors?.[0].message).to.equal(ACCESS_DENIED_ERROR);
+      expect(result.errors?.[0].extensions?.code).to.equal('UNAUTHENTICATED');
```

Comment:
Trying to check for the error code where possible. It's possible here but I remember we had issues before with other error codes returned by `apollo-server-errors`.

Reply:
This is nice, let me do that in my tests too